### PR TITLE
[Enhancement] Revise StableDiffusion and corresponding components

### PR DESCRIPTION
**Pull request by LeoXing1996**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Support more flexible initialization for `StableDiffusion`.

## Modification

1. Revise the initialization function for `StableDiffusion`
2. Revise the output type of `StableDiffusion.infer`
3. Add some necessary attribute and method for `DenoisingUnet` and `DDIMScheduler`.
4. Register MMEdit's `AutoencoderKL` as `EditAutoencoderKL` to support build from config.
5. Support a wrapper for diffusers scheduler to support load from pretrained.
6. Revise some docstring and remove some useless comments.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

**Before PR**:

- [x] I have read and followed the workflow indicated in the [CONTRIBUTING.md](https://github.com/open-mmlab/mmediting/blob/master/.github/CONTRIBUTING.md) to create this PR.
- [x] Pre-commit or linting tools indicated in [CONTRIBUTING.md](https://github.com/open-mmlab/mmediting/blob/master/.github/CONTRIBUTING.md) are used to fix the potential lint issues.
- [x] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [x] New functionalities are covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [x] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [x] If the modification has potential influence on downstream or other related projects, this PR should be tested with some of those projects.
- [x] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

**Comment by liuwenran** [comment]
can we run stable diffusion like before?

```python
from mmengine import MODELS, Config
from torchvision import utils

from mmedit.utils import register_all_modules

register_all_modules()

config = 'configs/stable_diffusion/stable-diffusion_ddim_denoisingunet.py'
config = Config.fromfile(config).copy()
config.model.init_cfg.pretrained_model_path = '/path/to/your/stable-diffusion-v1-5'

StableDiffuser = MODELS.build(config.model)
prompt = 'A mecha robot in a favela in expressionist style'
StableDiffuser = StableDiffuser.to('cuda')

image = StableDiffuser.infer(prompt)['samples']
utils.save_image(image, 'robot.png')
```

File: configs/stable_diffusion/stable-diffusion_ddim_denoisingunet.py [hunk 1]
```diff
@@ -47,12 +48,15 @@
...
     unet=unet,
     vae=vae,
-    init_cfg=init_cfg,
-)
+    text_encoder=dict(
+        type='ClipWrapper',
+        clip_type='huggingface',
+        pretrained_model_name_or_path='runwayml/stable-diffusion-v1-5',
+        subfolder='text_encoder'),
+    tokenizer='runwayml/stable-diffusion-v1-5',
```
Comment (liuwenran) [inline configs/stable_diffusion/stable-diffusion_ddim_denoisingunet.py:60]: why tokenizer is directly from runwayml but text_encoder needs a clipwrapper?

Comment (LeoXing1996) [reply configs/stable_diffusion/stable-diffusion_ddim_denoisingunet.py:60]: This is a good idea, I will support a wrapper for the tokenizer in textual inversion.

**Comment by LeoXing1996** [comment]
> can we run stable diffusion like before?
> 
> ```python
> from mmengine import MODELS, Config
> from torchvision import utils
> 
> from mmedit.utils import register_all_modules
> 
> register_all_modules()
> 
> config = 'configs/stable_diffusion/stable-diffusion_ddim_denoisingunet.py'
> config = Config.fromfile(config).copy()
> config.model.init_cfg.pretrained_model_path = '/path/to/your/stable-diffusion-v1-5'
> 
> StableDiffuser = MODELS.build(config.model)
> prompt = 'A mecha robot in a favela in expressionist style'
> StableDiffuser = StableDiffuser.to('cuda')
> 
> image = StableDiffuser.infer(prompt)['samples']
> utils.save_image(image, 'robot.png')
> ```

`config.model.init_cfg.pretrained_model_path = '/path/to/your/stable-diffusion-v1-5'` can be removed, since pretrained URL for each submodules has been defined in configs.

**Review by plyfager — APPROVED** [review]

**Comment by plyfager** [comment]
> can we run stable diffusion like before?
> 
> ```python
> from mmengine import MODELS, Config
> from torchvision import utils
> 
> from mmedit.utils import register_all_modules
> 
> register_all_modules()
> 
> config = 'configs/stable_diffusion/stable-diffusion_ddim_denoisingunet.py'
> config = Config.fromfile(config).copy()
> config.model.init_cfg.pretrained_model_path = '/path/to/your/stable-diffusion-v1-5'
> 
> StableDiffuser = MODELS.build(config.model)
> prompt = 'A mecha robot in a favela in expressionist style'
> StableDiffuser = StableDiffuser.to('cuda')
> 
> image = StableDiffuser.infer(prompt)['samples']
> utils.save_image(image, 'robot.png')
> ```

Maybe we can add a class method ``from_pretrained`` in a separate PR.

**Pull request merged**